commit 887318616cb662e8b283e1dcea0f00bee1589420
Author: Eike Hein <hein@kde.org>
Date:   Tue Nov 5 21:12:34 2013 +0100

    Always allow dragging tasks (but reorder only in manual sort mode).
    
    This means you don't need to enable manual sorting just to gain the
    ability to e.g. drag tasks onto the pager.
    
    Drags onto other tasks are ignored for spring loading however, to
    avoid annoyance in just that use case (and since it's not very use-
    ful to begin with, unlike dragging document content onto a task).

diff --git a/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml b/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
index 53b5d89..03da0b2 100644
--- a/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
+++ b/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
@@ -42,14 +42,14 @@ Item {
 
             var above = target.childAt(event.x, event.y);
 
-            if (tasks.dragSource) {
+            if (tasks.dragSource && tasks.manualSorting) {
                 if (tasks.dragSource != above && !tasks.dragSource.isLauncher
                     && !(above && "isLauncher" in above && above.isLauncher)) {
                     itemMove(tasks.dragSource.itemId,
                         TaskTools.insertionIndexAt(tasks.dragSource.itemIndex,
                             event.x, event.y));
                 }
-            } else if (above && hoveredItem != above) {
+            } else if (!tasks.dragSource && above && hoveredItem != above) {
                 hoveredItem = above;
                 activationTimer.start();
             } else if (!above) {
diff --git a/plasma/desktop/applets/tasks/package/contents/ui/Task.qml b/plasma/desktop/applets/tasks/package/contents/ui/Task.qml
index 8c268af..0fe1a7d 100644
--- a/plasma/desktop/applets/tasks/package/contents/ui/Task.qml
+++ b/plasma/desktop/applets/tasks/package/contents/ui/Task.qml
@@ -124,7 +124,7 @@ MouseEventListener {
     }
 
     onPositionChanged: {
-        if (tasks.manualSorting && pressX != -1 && dragHelper.isDrag(pressX, pressY, mouse.x, mouse.y)) {
+        if (pressX != -1 && dragHelper.isDrag(pressX, pressY, mouse.x, mouse.y)) {
             tasks.dragSource = task;
             dragHelper.startDrag(model.MimeType, model.MimeData, model.LauncherUrl, model.DecorationRole);
             pressX = -1;
